pkg/apis/secretmanager/v1alpha1/secretstore_gcp_types.go

```diff
@@ -0,0 +1,44 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import smmeta "github.com/itscontained/secret-manager/pkg/apis/meta/v1"
+
+// Configures an store to sync secrets using GCP Secret Manager
+type GCPStore struct {
+	// ProjectID is a convenience string to allow the shortening of secret paths.
+	// When set, the prefix projects/<ProjectID> can be removed from the name
+	ProjectID *string `json:"projectID,omitempty"`
+	// Auth configures how secret-manager authenticates with GCP Secret Manager.
+	// +optional
+	AuthSecretRef *GCPAuth `json:"authSecretRef,omitempty"`
+}
+
+// Configuration used to authenticate with GCP.
+// Either of `JSON` or `File` can be specified. If not set we fall-back to using
+// `GOOGLE_APPLICATION_CREDENTIALS` or the default service account of the compute engine
+// see: https://cloud.google.com/docs/authentication/production
+type GCPAuth struct {
+	// The JSON secret key selector is used for authentication. If not set we fall-back to using
+	// `GOOGLE_APPLICATION_CREDENTIALS` or the default service account of the compute engine
+	// see: https://cloud.google.com/docs/authentication/production
+	// +optional
+	JSON *smmeta.SecretKeySelector `json:"json,omitempty"`
+	// The File string is used for authentication using a gcp credentials json file. If not set we fall-back to using
+	// `GOOGLE_APPLICATION_CREDENTIALS` or the default service account of the compute engine
+	// see: https://cloud.google.com/docs/authentication/production
+	// +optional
+	File *string `json:"file,omitempty"`
```

Comment:
Thoughts on `filePath` to be more specific here. 

I'm also not sure this setting is needed as this already requires configuration of the controller to have the credential file mounted, feels like we are better off just letting them use default file location or ENV configuration than allowing overriding file location.

Reply:
I guess to be more specific, configuring this requires also configuring the controller to already have the file at this location, and tightly couples controller and store. At that point it seems more natural to just use cloud provider configuration than overriding it in the store.

Reply:
Renamed.

Reference: Decided in discord that existence is plausible in the event they want unique credentials for multiple stores mounted.